src/modulefinder/sentry_modulefinder_linux.c

```diff
@@ -315,15 +371,15 @@ get_code_id_from_text_fallback(const sentry_module_t *module)
             elf.e_shoff + elf.e_shentsize * elf.e_shstrndx,
             sizeof(Elf64_Shdr)));
 
-        const char *names = sentry__module_get_addr(
-            module, strheader.sh_offset, strheader.sh_entsize);
-        ENSURE(names);
         for (int i = 0; i < elf.e_shnum; i++) {
             Elf64_Shdr header;
             ENSURE(sentry__module_read_safely(&header, module,
                 elf.e_shoff + elf.e_shentsize * i, sizeof(Elf64_Shdr)));
 
-            const char *name = names + header.sh_name;
+            char name[6];
+            ENSURE(sentry__module_read_safely(name, module,
+                strheader.sh_offset + header.sh_name, sizeof(name)));
+            name[5] = '\0';
```

Comment:
to make reviewing easier it is nicer if you set the base to the other branches this started from.  once those branches are merged github will automatically change the base to master and this can then be merged.